### ISSUE TITLE
Compiling Test harness - IntegerOverflow

### ISSUE THREAD

**Issue by akhikolla** [trailofbits/deepstate#344]
How should be the format of the compile line to compile any of the example programs in deepstate.
For suppose when I compile IntegerOverflow using make file it compiles.
But when I see what is the command that is being used to compile the example, all I see are nested makefiles and has the following lines to compile IntegerOverflow:

```
#=============================================================================
# Target rules for targets named IntegerOverflow

# Build rule for target.
IntegerOverflow: cmake_check_build_system
	$(MAKE) -f CMakeFiles/Makefile2 IntegerOverflow
.PHONY : IntegerOverflow

# fast build rule for target.
IntegerOverflow/fast:
	$(MAKE) -f examples/CMakeFiles/IntegerOverflow.dir/build.make examples/CMakeFiles/IntegerOverflow.dir/build
.PHONY : IntegerOverflow/fast
#=============================================================================
# Target rules for target examples/CMakeFiles/IntegerOverflow.dir

# All Build rule for target.
examples/CMakeFiles/IntegerOverflow.dir/all: CMakeFiles/deepstate.dir/all
	$(MAKE) -f examples/CMakeFiles/IntegerOverflow.dir/build.make examples/CMakeFiles/IntegerOverflow.dir/depend
	$(MAKE) -f examples/CMakeFiles/IntegerOverflow.dir/build.make examples/CMakeFiles/IntegerOverflow.dir/build
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --progress-dir=/home/akhila/Desktop/deepstate/build/CMakeFiles --progress-num=17,18 "Built target IntegerOverflow"
.PHONY : examples/CMakeFiles/IntegerOverflow.dir/all

# Include target in all.
all: examples/CMakeFiles/IntegerOverflow.dir/all

.PHONY : all

# Build rule for subdir invocation for target.
examples/CMakeFiles/IntegerOverflow.dir/rule: cmake_check_build_system
	$(CMAKE_COMMAND) -E cmake_progress_start /home/akhila/Desktop/deepstate/build/CMakeFiles 7
	$(MAKE) -f CMakeFiles/Makefile2 examples/CMakeFiles/IntegerOverflow.dir/all
	$(CMAKE_COMMAND) -E cmake_progress_start /home/akhila/Desktop/deepstate/build/CMakeFiles 0
.PHONY : examples/CMakeFiles/IntegerOverflow.dir/rule

# Convenience name for target.
IntegerOverflow: examples/CMakeFiles/IntegerOverflow.dir/rule

.PHONY : IntegerOverflow

#=================================================
```

Can someone please explain me the flow or help me with a single line command to compile the code.

EDIT: I edited the post and moved the code to a code box (@disconnect3d).

**Comment by GrosQuildu** [comment]
First, build deepstate itself (see https://github.com/trailofbits/deepstate/tree/master#building-on-ubuntu-1804-bionic). It will produce library `libdeepstate.a`.

Then, compile example program as you would standard C/C++ program, just add `-ldeepstate` option when linking.

We probably should mention `libdeepstate.a` in the README @agroce

**Comment by akhikolla** [comment]
I made a few changes to IntegerOverflow and changed it to something like this:
I am trying to integrate Deepstate with Rcpp, and wrote following fuction. When I run my code as you said i.e
./IntegerOverflow -ldeepstate

it gives the the following . 
![Screenshot from 2020-03-09 13-47-54](https://user-images.githubusercontent.com/31878891/76256671-9315d580-620d-11ea-8292-15ba3b33bbbc.png)

I did compile my code and started running the following, do you think is this right format to integrate Rcpp with deepstate. Please help me.

#include <deepstate/DeepState.hpp>
#include <Rcpp.h>
using namespace deepstate;
using namespace Rcpp;

// [[Rcpp::export]]
Rcpp::NumericVector timesTwo(NumericVector x) {
  return x * 2;
}

TEST(SignedInteger, AdditionOverflow) {
  int x;
  x = DeepState_RandInt();
  timesTwo(x);
}

I dont think its giving me the output that I expected.

**Comment by agroce** [comment]
What output are you expecting?

**Comment by akhikolla** [comment]
It should just pass the random inputs generated from the RandInt() function to timesTwo.
Showing random fuzzing inputs.

I didn't use Multiplication Overflow test case in the above code I forwarded but when I compiled and run the code using **./IntegerOverflow -ldeepstate** why is it showing 
**TRACE: RUNNING SignedInteger_MultiplicationOverflow** 
Is this because the test cases are already stored? If so how can I compile the updated code without using running old test cases?
